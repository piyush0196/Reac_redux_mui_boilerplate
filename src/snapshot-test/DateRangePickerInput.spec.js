import { theme } from 'theme/theme'
import { ThemeProvider } from '@mui/material'
import TestRenderer from 'react-test-renderer'
import DateRangePickerInput from 'components/ui-kit/DateRangePickerInput'

describe('Component - DateRangePickerInput', () => {
  const wrappers = []
  const types = ['default', 'dateFilter']
  const DateRangePickerInputJSX = ({ type }) => (
    <ThemeProvider theme={theme}>
      <DateRangePickerInput title='demo title' type={type} />
    </ThemeProvider>
  )

  types.map(type => (
    wrappers.push(TestRenderer.create(
      <DateRangePickerInputJSX type={type} />
    ).toJSON())
  ))

  it('renders all type of DateRangePickerInput component correctly', () => {
    wrappers.forEach(wrapper => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
