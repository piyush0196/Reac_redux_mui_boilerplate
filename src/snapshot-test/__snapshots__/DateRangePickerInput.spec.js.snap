// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DateRangePickerInput renders all type of DateRangePickerInput component correctly 1`] = `
<div
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="css-17vdz66"
  >
    <div
      style={
        Object {
          "height": "100px",
          "width": "fit-content",
        }
      }
    >
      <div
        style={
          Object {
            "textAlign": "start",
          }
        }
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
        >
          demo title*
        </h6>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-8qan1k-MuiFormControl-root-MuiTextField-root"
        labeltype="default"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1gzianm-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
            name=""
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            placeholder="mm/dd/yyyy"
            required={false}
            type="tel"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-nnbavb"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component - DateRangePickerInput renders all type of DateRangePickerInput component correctly 2`] = `
<div
  style={
    Object {
      "width": "300px",
    }
  }
>
  <div
    className="css-17vdz66"
  >
    <div
      className="MuiBox-root css-i9rics"
      type="dateFilter"
    >
      <div
        style={
          Object {
            "marginRight": "10px",
            "width": "max-content",
          }
        }
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-5ts02r-MuiTypography-root"
        >
          demo title
        </h6>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1bptqyh-MuiFormControl-root-MuiTextField-root"
        labeltype="dateFilter"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-1gzianm-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
            id="mui-3"
            name=""
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            placeholder="mm/dd/yyyy"
            required={false}
            type="tel"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-nnbavb"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
